lua/nvim-tree/actions/git/clipboard.lua

```diff
@@ -0,0 +1,96 @@
+local M = {}
+
+--- Bag / Set like class to keep uniq nodes
+--- @class gitClipboard
+--- @field prototype gitClipboardInstance
+--- @field prototype.constructor gitClipboard
+M.Clipboard = { prototype = { constructor = M.Clipboard } }
```

Comment:
The prototype pattern is surfacing a lot of warnings and confusing lua_language_server. Any reason we need that over an "ordinary" class?